### DISCUSSION TITLE
Linux Kernel 5.16正式版发布，为作者避雷

### DISCUSSION THREAD

**Discussion by NekoCompiler** [jinwyp/one_click_script#110] [General]
（以下讨论不包含CentOS宇宙，只包含 Ubuntu Kernel Mainline，CentOS类系统刷内核一般用elrepo仓库直接刷就好）

kernel.org 和 Ubuntu Kernel Mainline 均已发布 kernel 5.16 主线版

刷 5.16 内核要安装 Ubuntu Kernel Mainline 的这些deb软件包：

linux-headers-all
linux-headers-generic-amd64
linux-image-generic-amd64
linux-modules-generic-amd64

但是，有两个值得关注的问题

第一，5.16内核的deb软件包采用zstd打包，目前 Debian 11 的dpkg软件包安装器暂不支持这种软件包，所以 Debian 11 刷不了，只能由 Ubuntu 20.04 LTS 或更高版本享用。

Debian 11 目前最多可以刷到 Ubuntu Kernel Mainline 的5.15.13，需要Debian实验性软件仓库（Experimental）的最新libc6和libssl3。

第二，5.16内核依赖最新的libc6和libssl3，注意是最新的，就类似于开发版那种，通俗易懂地说就是依赖 Ubuntu 22.04 LTS 版本的libc6和libssl3，非常新。

但是解法很简单，那就是 Ubuntu 20.04 LTS 直接安装 Ubuntu 22.04 LTS 的libc6和libssl3软件包就可以了。

用wget下载这些在 Ubuntu 22.04 LTS 仓库中的软件包：

libc6

https://packages.ubuntu.com/jammy/libc6 （主页）

```
http://mirrors.kernel.org/ubuntu/pool/main/g/glibc/libc6_2.34-0ubuntu3_amd64.deb
```

libssl3

https://packages.ubuntu.com/jammy/libssl3 （主页）

```
http://mirrors.kernel.org/ubuntu/pool/main/o/openssl/libssl3_3.0.0-1ubuntu1_amd64.deb
```

locales

https://packages.ubuntu.com/jammy/locales （主页）

```
http://mirrors.kernel.org/ubuntu/pool/main/g/glibc/locales_2.34-0ubuntu3_all.deb
```

libc-bin

https://packages.ubuntu.com/jammy/libc-bin （主页）

```
http://mirrors.kernel.org/ubuntu/pool/main/g/glibc/libc-bin_2.34-0ubuntu3_amd64.deb
```

安装
```
dpkg -i libc-bin_2.34-0ubuntu3_amd64.deb locales_2.34-0ubuntu3_all.deb libc6_2.34-0ubuntu3_amd64.deb libssl3_3.0.0-1ubuntu1_amd64.deb
```

安装好之后重启，然后直接安装 Ubuntu Kernel Mainline 的4个包就可以了，安装好之后再重启。

![IMG_20220110_200136.jpg](https://user-images.githubusercontent.com/62339707/148769124-794ee763-5fdb-44b7-84dc-d8f0c9351e3e.jpg)

**Comment by jinwyp** [comment]
这个问题已经解决 支持debian和ubuntu 升级5.16和5.17. 可以试试

**Comment by wnark** [comment]
安装的时候会报错
```
root@linux:~/tmp/deb# dpkg -i libc6_2.35-0ubuntu3_amd64.deb 
dpkg: 关于含有 libc6:amd64 的 libc6_2.35-0ubuntu3_amd64.deb：
 libc6:amd64 会破坏 fakeroot (<< 1.25.3-1.1ubuntu2~)
  fakeroot (版本为 1.24-1)已存在并且 已安装。

dpkg: 处理归档 libc6_2.35-0ubuntu3_amd64.deb (--install)时出错：
 安装 libc6:amd64 将破坏 fakeroot，并且
不允许反配置(--auto-deconfigure 也许会有帮助)
在处理时有错误发生：
 libc6_2.35-0ubuntu3_amd64.deb
root@linux:~/tmp/deb# dpkg -i libc6_2.35-0ubuntu3_amd64.deb --auto-deconfigure
dpkg: 错误: 无法访问归档 '--auto-deconfigure': 没有那个文件或目录
```

**Reply by wnark** [reply]
在笔记本上Ubuntu20.04尝试安装jammy版本的fakeroot等组件，重启后Ubuntu黑屏无法使用，这种方法不一定可行
